scylla-cql/src/frame/types.rs

```diff
@@ -98,6 +98,12 @@ impl From<std::str::Utf8Error> for ParseError {
     }
 }
 
+impl From<Infallible> for ParseError {
+    fn from(_: Infallible) -> Self {
+        ParseError::BadIncomingData("Unexpected Infallible Error".to_string())
+    }
+}
```

Comment:
Could you explain what this trait implementation does?
AFAIU `Infallible` is for things that can never happen, so why do we want to convert it to a `ParseError`?

https://doc.rust-lang.org/std/convert/enum.Infallible.html

Reply:
It's for [the conversion ](https://github.com/scylladb/scylla-rust-driver/blob/8dfdcdbc6ea7299b76ee79e625f4a4f8cd009f15/scylla-cql/src/frame/request/batch.rs#L193)of `u16` to a `usize`. I wanted to do a simple `as` but didn't want to modify the existing code as much. I think converting from the previous `i16` to a `usize` would have failed with the `TryFromIntError` error, but with `u16` -> `usize`, it really is in fact infallible

Reply:
Ah okay, I understand now.
So we have a few pieces of code like this one:
```rust
let a: i16 = 123;
let b: usize = a.try_into()?;
```
And after changing the `i16` to `u16` they no longer compile because `try_into()` has an `Infallible` error type.
We can implement a conversion from `Infallible` to `ParseError` to make it compile, but it's a bit hacky.

I think it would be better to replace the `try_into()?`with `into()`, like this:
```rust
let a: u16 = 123;
let b: usize = a.into();
```

There is a an implentation of `From<u16> for usize`, so we can just use `into()` here.